llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -150,6 +150,11 @@ static cl::opt<bool> UseLIRCodeSizeHeurs(
              "with -Os/-Oz"),
     cl::init(true), cl::Hidden);
 
+static cl::opt<bool> ForceMemsetPatternIntrinsic(
```

Comment:
I would mildly suggest dropping this cl::opt from the final patch.  We probably should have a generic means to force enable a TLI function, but having one only for memset.pattern seems slightly odd.  Not a strongly held view, will leave to your discretion.

Reply:
I do have a patch somewhere that adds an opt flag to do this

Reply:
https://github.com/llvm/llvm-project/pull/145808